### PR TITLE
fix: Respect `glob=False` for cloud reads

### PR DESCRIPTION
**Pull request by nameexhaustion**
Fixes https://github.com/pola-rs/polars/issues/17822

### CODE REVIEW

File: crates/polars-utils/src/io.rs [hunk 1]
```diff
@@ -4,9 +4,13 @@ use std::path::Path;
...
 use polars_error::*;
 
+fn verbose() -> bool {
+    std::env::var("POLARS_VERBOSE").as_deref().unwrap_or("") == "1"
+}
+
 pub fn _limit_path_len_io_err(path: &Path, err: io::Error) -> PolarsError {
     let path = path.to_string_lossy();
-    let msg = if path.len() > 88 {
+    let msg = if path.len() > 88 && !verbose() {
```
Comment (nameexhaustion) [inline crates/polars-utils/src/io.rs:13]: drive-by QOL - print the full path if `POLARS_VERBOSE` is set

File: py-polars/tests/unit/io/test_other.py [hunk 2]
```diff
@@ -122,3 +122,34 @@ def test_unit_io_subdir_has_no_init() -> None:
...
+@pytest.mark.write_disk()
+@pytest.mark.parametrize(
+    ("scan_funcs", "write_func"),
+    [
+        ([pl.scan_parquet, pl.read_parquet], pl.DataFrame.write_parquet),
+        ([pl.scan_csv, pl.read_csv], pl.DataFrame.write_csv),
+    ],
+)
+@pytest.mark.parametrize("char", ["[", "*", "?"])
+def test_no_glob(
```
Comment (nameexhaustion) [inline py-polars/tests/unit/io/test_other.py:136]: consolidated all the tests here

**Review by ritchie46 — APPROVED** [review]
